scheduler/scheduler_system_test.go

```diff
@@ -2132,3 +2132,410 @@ func TestSystemSched_canHandle(t *testing.T) {
 		require.False(t, s.canHandle(structs.EvalTriggerPeriodicJob))
 	})
 }
+
+func TestSystemSched_NodeDisconnected(t *testing.T) {
```

Comment:
I don't think this test covers any cases where the client disconnects but the alloc does not have `MaxClientDisconnect`, does it? Our tests should make sure we're not breaking the existing behavior.

Reply:
It currently only covers lost / down nodes in terms of existing behavior. I'll add some more and try to get the function covered.